vscode-dotnet-runtime-library/src/Acquisition/WinMacGlobalInstaller.ts

```diff
@@ -349,11 +349,25 @@ If you were waiting for the install to succeed, please extend the timeout settin
         }
     }
 
-    private getMacPath() : string
+    private getMacPath(macPathShouldExist = true) : string
     {
-        // On an arm machine we would install to /usr/local/share/dotnet/x64/dotnet/sdk` for a 64 bit sdk
-        // but we don't currently allow customizing the install architecture so that would never happen.
-        return path.resolve(`/usr/local/share/dotnet/dotnet`);
+        const standardHostPath = path.resolve(`/usr/local/share/dotnet/dotnet`);
+        const arm64EmulationHostPath = path.resolve(`/usr/local/share/dotnet/x64/dotnet`);
+        if(!macPathShouldExist)
+        {
+            return standardHostPath;
+        }
+        else
+        {
+            if(fs.existsSync(standardHostPath) || !fs.existsSync(arm64EmulationHostPath))
+            {
+                return standardHostPath;
+            }
+            else
+            {
+                return arm64EmulationHostPath;
+            }
+        }
```

Comment:
Isn't this just:
```
        if(!macPathShouldExist || fs.existsSync(standardHostPath) || !fs.existsSync(arm64EmulationHostPath))
        {
            return standardHostPath;
        }
        else
        {
            return arm64EmulationHostPath;
        }
```

And if that is the case, this would be simpler:
```suggestion
        return !macPathShouldExist ||
        fs.existsSync(standardHostPath) ||
        !fs.existsSync(arm64EmulationHostPath)) ?
            standardHostPath : 
            arm64EmulationHostPath;
```

Or you can make it one long line or whatever format you like.

Reply:
You're right, this is much better 😉  Thank you. Sometimes  I dont think about simplifying code in this way.